New hero

vikShevchenko opened the pull request.

vikShevchenko committed:
Update _hero.scss

vikShevchenko committed:
Update hero.html

vikShevchenko committed:
upload

vikShevchenko committed:
update 1280

KaterynaVeselova commented on src/partials/hero.html:
```diff
@@ -0,0 +1,35 @@
+
+      <div class="hero__foto--top">
+
+        <div class="hero__slider--but-l">
```
тут би краще сробити тег button замість div (не семантично)

vikShevchenko replied:
+

KaterynaVeselova commented on src/partials/hero.html:
```diff
@@ -0,0 +1,35 @@
+          </svg>
+        </div>
+      <picture>
+        <source srcset="./images/hero/hero1_1200@1x.png 1x, ./images/hero/hero1_1200@2x.png 2x" media="(min-width: 1280px)"/>
```
давайте погодимо використовувати в медіа правилах змінні на breakpoints (в даному випадку min-width: $desktop)

vikShevchenko replied:
ok. А де ця змінна, щоб використати?

KaterynaVeselova replied:
![image](https://user-images.githubusercontent.com/107804176/187068828-18bd6580-15cd-4e0b-8d58-f0815025d23c.png)

KaterynaVeselova replied:
пишете за зразком 
@media screen and (min-width: $tablet){
}

KaterynaVeselova commented on src/partials/hero.html:
```diff
@@ -0,0 +1,35 @@
+      <picture>
+        <source srcset="./images/hero/hero1_1200@1x.png 1x, ./images/hero/hero1_1200@2x.png 2x" media="(min-width: 1280px)"/>
+        <source srcset="./images/hero/hero1_768@1x.png 1x, ./images/hero/hero1_768@2x.png 2x" media="(min-width: 768px)"/>
+        <img class="hero__foto--size" src="./images/hero/hero1_1200@1x.png" alt="Фото блюда"/>
```
по дефолту краще поставити картинку мінімального розміру (тут краще використати 768@1x)

vikShevchenko replied:
+

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+    align-items: center;
+    position: relative;
+
+    @media screen  and (min-width: 320px){
```
навіщо тут медіа правило, просто flex-direction: column; без правила

vikShevchenko replied:
?

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        
+    }
+    @media screen  and (min-width: 1280px){
+        display: flex;
```
тут , це вже повторення, є вище: можна просто залишити flex-direction: row; решта зверху має підтягнутися та justify-content: space-between;

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+
+.hero__general--top{
+    display: flex;
+    justify-content: space-between;
```
оцей justify-content: space-between; має застосовуватися тільки в медіа правилі з @media screen  and (min-width: 1280px)

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+    @media screen  and (min-width: 320px){
+        flex-direction: column;
+        
+    }
```
а на планшетний варіант, я би картинці задала позиціонування абсолютом відносно вашої секції, якій ви вже раніше надали позиціонування relative

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        
+    }
+    @media screen  and (min-width: 768px){
+        width: 704px;
```
наче 537px там ширина цього div

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+.hero__title--top{
+    color: var(--main-text-color);  
+    @media screen  and (min-width: 320px){
+        margin-top: 60px;
```
margin  використовуємо лише для відступів між елементами в середині секції. Тут або padding тоді вже, або top: -60px; оскільки позиціонування relative.

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+    @media screen  and (min-width: 320px){
+        margin-top: 60px;
+        font-size: 28px;
+        line-height: 33px;   
```
line-height це 33/28, а не 33px

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        line-height: 33px;   
+    } 
+    @media screen  and (min-width: 768px){
+        width: 537px;
```
width дублюєте, було на контейнері, просто там ви невірно вказали, я поправила, краще тоді width: 100%

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        font-size: 48px;
+        line-height: 120%;
+        margin-top: 0px;
+        width: 503px;
```
тут теж width дублюєте, було на контейнері, краще тоді width: 100%

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        font-weight: 400;
+        font-size: 16px;
+        line-height: 150%;
+        width: 503px; 
```
є на батьківському width: 100%
фонти переписуєте однакові стилі на таблетці достаньо, вже в десктопі можна не писати, вони однакові

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+    position: relative;
+    border-radius: 50%;
+    
+    @media screen  and (min-width: 320px){
```
@media screen  and (max-width: 767px){

а якщо попередній комент поправити, то медіа правило взагалі не потрібно, просто display: none;

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+}
+.hero__foto--top{
+    position: relative;
+    border-radius: 50%;
```
position: relative; - навіщо це тут?
border-radius: 50%; - це краще перенести до таблетки вже

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        border: 14px solid var(--yellow-main);
+        
+        position: absolute;
+        top: 135px;
```
та наче 150

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        
+        position: absolute;
+        top: 135px;
+        left: 370px;
```
а тут 400, хоча я не розумію, звідки ти брав ці значення

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        
+        position: relative;
+        border: 26px solid var(--yellow-main);
+        margin-top: 16px;
```
для чого цей маржин?

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        position: relative;
+        border: 26px solid var(--yellow-main);
+        margin-top: 16px;
+        margin-right: 14px;
```
і взагалі, хіба маржини так працюють при позиціонуванні вийнятого з потоку елемента?

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+
+    }
+}
+.hero__foto--size{
```
цей клас взагалі для картинки по дефолту, навіщо його описувати?

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+.hero__foto--size{
+    width: 580px;
+}
+.hero__slider--but-l{
```
оці кнопки однакові, краще їм один спільний клас дати , наприклад slider__btn та надати поруч клас-модифікатор до правої та лівої кнопки , наприклад slider__btn--left та slider__btn--right.
загальному класу надати загальні стилі а вже top та right при позиціонуванні кожному свої, бо ти дублюєш багато правил.

KaterynaVeselova replied:
а взагалі я так і не зрозуміла, відносно чого ти позиціонуєш кнопки для таблетки, якщо у того діва позиція теж absolute

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+
+.hero__btn{
+    background: var(--btn-default);
+    font-weight: 400;
```
спадкує від body

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+    background: var(--btn-default);
+    font-weight: 400;
+    font-size: 14px;
+    line-height: 150%;
```
спадкує від body

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+        
+    }
+    @media screen  and (min-width: 1280px){
+        margin-bottom: 0px;
```
margin-bottom: 164px;

vikShevchenko replied:
+?

KaterynaVeselova commented on src/sass/layout/_hero.scss:
```diff
@@ -0,0 +1,221 @@
+    width: 100px;
+    text-align: center;
+    @extend %btn;
+    @media screen  and (min-width: 320px){
```
@media screen  and (min-width: $tablet){
          margin-bottom: 738px;
}
А от у кнопки планшета немає марджинів. Є падінги, але на секцію: нижній - 50px, верхній - 60px

vikShevchenko replied:
+?

KaterynaVeselova reviewed (commented):
Вікторе, з позиціонуванням каруселі я зовсім заплуталась, моя логіка розірвана - "підпис здесь і підпис здесь". Почитай коменти, але потрібно щоб Інга завтра глянула

vikShevchenko committed:
upload

Треба "грохнуть"  fill-и у sprite.svg
стр 22 26
Я їх перебити не можу при ховері.

KaterynaVeselova commented on src/partials/hero.html:
```diff
@@ -13,21 +13,24 @@ <h1 class="hero__title--top">Enjoy Your Favorite Food From Anywhere</h1>
       <picture>
-        <source srcset="./images/hero/hero1_1200@1x.png 1x, ./images/hero/hero1_1200@2x.png 2x" media="(min-width: 1280px)"/>
+        <source srcset="./images/hero/hero1_1200@1x.png 1x, ./images/hero/hero1_1200@2x.png 2x" media="(min-width: $desktop)"/>
         <source srcset="./images/hero/hero1_768@1x.png 1x, ./images/hero/hero1_768@2x.png 2x" media="(min-width: 768px)"/>
```
media="(min-width: $tablet)"
